src/sdk/accountspace_fs.js

```diff
@@ -259,73 +231,233 @@ class AccountSpaceFS {
     // ACCESS KEY //
     ////////////////
 
+    // 1 - check that the requesting account is a root user account or that the username is same as the requester
+    // 2 - check that the requested account config file exists
+    // 3 - read the account config file (and decrypt its existing encrypted secret keys and then encrypted secret keys)
+    // 4 - if the requester is root user account - check that it owns the account
+    //     check that the access key to create is on a user is owned by the the root account
+    // 5 - check that the number of access key array
+    // 6 - generate access keys
+    // 7 - encryption
+    // 8 - validate account
+    // 9 - update account config file
+    // 10 - link new access key file to config file
     async create_access_key(params, account_sdk) {
-        const { dummy_access_key } = get_user_details(params.username);
-        dbg.log1('create_access_key', params);
-        return {
-            username: dummy_access_key.username,
-            access_key: dummy_access_key.access_key,
-            status: dummy_access_key.status,
-            secret_key: dummy_access_key.secret_key,
-            create_date: new Date(),
-        };
+        const action = 'create_access_key';
+        dbg.log1(`AccountSpaceFS.${action}`, params, account_sdk);
+        try {
+            const requesting_account = account_sdk.requesting_account;
+            const requester = this._check_if_requesting_account_is_root_account_or_user_om_himself(action,
+                requesting_account, params.username);
+            const name_for_access_key = params.username ?? requester.name;
```

Comment:
this ?? operator will only fallback if undefined, but not if empty string. is that really what you meant??

Reply:
I think using the ?? operator is quite esoteric and are probably not intended for common cases.
It is mostly useful to avoid number 0 from being considered as falsy value (much less useful for empty strings or boolean false values...)